### PR TITLE
Press0 1453 dark mode preview

### PR DESCRIPTION
**Pull request by ajayadav09**
## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by officiallygod — COMMENTED** [review]
This looks good but should we clear the localStorage values created during onboarding? As, I suppose these might not be useful later and might take up memory.

Also, considering we use strings like `'nfd-sitegen-theme-mode'` very often should we go about using constants? or helper methods to set data or retrieve data like we do using **Options** and **get_option_name** in Php?

**Comment by ajayadav09** [comment]
> This looks good but should we clear the localStorage values created during onboarding? As, I suppose these might not be useful later and might take up memory.
> 
> Also, considering we use strings like `'nfd-sitegen-theme-mode'` very often should we go about using constants? or helper methods to set data or retrieve data like we do using **Options** and **get_option_name** in Php?

@officiallygod  @arunshenoy99  moved from localstorage to store and saving the theme value in onboarding data here

https://github.com/newfold-labs/wp-module-onboarding-data/pull/101

File: src/OnboardingSPA/components/ThemeContextProvider/index.js [hunk 1]
```diff
@@ -1,28 +1,54 @@
...
+	const { currentData } = useSelect( ( select ) => {
+		return {
+			currentData:
+				select( nfdOnboardingStore ).getCurrentOnboardingData(),
+		};
+	} );
+
+	if ( currentData.sitegenThemeMode ) {
+		return currentData.sitegenThemeMode;
+	}
```
Comment (officiallygod) [inline src/OnboardingSPA/components/ThemeContextProvider/index.js:19]: If this is '' does it fall here? Cause ig the default theme is the Device theme

File: src/OnboardingSPA/components/ThemeContextProvider/index.js [hunk 2]
```diff
@@ -1,28 +1,54 @@
...
 	) {
 		return THEME_DARK;
 	}
+
 	return THEME_LIGHT;
 }
 
 const ThemeProvider = ( { children } ) => {
-	const [ theme, setTheme ] = useState( getPreferredColorScheme );
+	const preferredColorScheme = usePreferredColorScheme();
```
Comment (officiallygod) [inline src/OnboardingSPA/components/ThemeContextProvider/index.js:32]: Considering here we have to call **usePreferredColorScheme** does it make sense to initialize `currentData` here and as well as in the above func or send as an param to the func

File: src/OnboardingSPA/components/ThemeContextProvider/index.js [hunk 3]
```diff
@@ -1,28 +1,54 @@
...
 		const colorSchemeMediaQuery = window.matchMedia(
 			'(prefers-color-scheme: dark)'
 		);
 
 		const handleChange = ( event ) => {
-			setTheme( event.matches ? THEME_DARK : THEME_LIGHT );
+			const newTheme = event.matches ? THEME_DARK : THEME_LIGHT;
+			setTheme( newTheme );
+			currentData.sitegenThemeMode = newTheme;
+			setFlow( currentData );
```
Comment (officiallygod) [inline src/OnboardingSPA/components/ThemeContextProvider/index.js:51]: Considering we are changing the data shouldn't we also do `setCurrentOnboardingData` and sync it to the store?

File: src/OnboardingSPA/components/ThemeContextProvider/index.js [hunk 4]
```diff
@@ -33,9 +59,13 @@ const ThemeProvider = ( { children } ) => {
 	}, [] );
 
 	const toggleTheme = () => {
-		setTheme( ( prevTheme ) =>
-			prevTheme === THEME_DARK ? THEME_LIGHT : THEME_DARK
-		);
+		setTheme( ( prevTheme ) => {
+			const newTheme =
+				prevTheme === THEME_DARK ? THEME_LIGHT : THEME_DARK;
+			currentData.sitegenThemeMode = newTheme;
```
Comment (officiallygod) [inline src/OnboardingSPA/components/ThemeContextProvider/index.js:65]: `setCurrentOnboardingData`

File: src/OnboardingSPA/store/reducer.js [hunk 5]
```diff
@@ -35,6 +35,7 @@ export function flow(
 		},
 		chapter: undefined,
 		interactionDisabled: false,
+		sitegenThemeMode: '',
```
Comment (officiallygod) [inline src/OnboardingSPA/store/reducer.js:38]: I doubt if this is required as we always have the data in currentData.sitegenThemeMode, then why here?

File: src/OnboardingSPA/components/ThemeContextProvider/index.js (hunk 1 above)
Comment (ajayadav09) [reply src/OnboardingSPA/components/ThemeContextProvider/index.js:19]: if the user toggles the theme, currentData.sitegenThemeMode will be set to that and if the user refreshes the page or revisits, the last theme will be applied.

**Review by officiallygod — APPROVED** [review]

**Comment by arunshenoy99** [comment]
PRESS0-1453

**Pull request merged**